pgmpy/factors/Factor.py

```diff
@@ -496,11 +496,24 @@ def __eq__(self, other):
         if type(self) != type(other):
             return False
 
-        if self.variables == other.variables and all(self.cardinality == other.cardinality) \
-                and all(self.values == other.values):
-            return True
-        else:
+        if set(self.scope()) != set(other.scope()):
             return False
+        else:
+            order_dict = {}
+            for var_index in range(len(self.scope())):
+                order_dict[var_index] = other.scope().index(self.scope()[var_index])
```

Comment:
@ankurankan In each iteration we are making a call to ```scope()``` method. Instead we should save the scope of the factor apriori and use it in each iteration and the same thing applies for ```other.scope()```.